src/canmatrix/compare.py

```diff
@@ -203,99 +218,104 @@ def compareDefineList(d1list, d2list):
             d2 = d2list[definition]
             d1 = d1list[definition]
             if d1.definition != d2.definition:
-                result.addChild(
-                    compareResult(
+                result.add_child(
+                    CompareResult(
                         "changed", "Definition", d1.definition, [
                             d1.definition, d2.definition]))
 
             if d1.defaultValue != d2.defaultValue:
-                result.addChild(
-                    compareResult(
+                result.add_child(
+                    CompareResult(
                         "changed", "DefaultValue", d1.definition, [
                             d1.defaultValue, d2.defaultValue]))
     for definition in d2list:
         if definition not in d1list:
-            result.addChild(
-                compareResult(
+            result.add_child(
+                CompareResult(
                     "added",
                     "Define" +
                     str(definition),
                     d2list))
     return result
 
 
-def compareAttributes(ele1, ele2, ignore=None):
-    result = compareResult("equal", "ATTRIBUTES", ele1)
-    if ignore is not None and "ATTRIBUTE" in ignore and (
+def compare_attributes(ele1, ele2, ignore=None):
+    # type: (typing.Union[cm.CanMatrix, cm.Ecu, cm.Frame, cm.Signal], typing.Union[cm.CanMatrix, cm.Ecu, cm.Frame, cm.Signal], typing.Optional[typing.Mapping[str, typing.Union[str, bool]]]) -> CompareResult
```

Comment:
We definitely need the union to define types, which have `attribute`. I'm not sure how to make "restricted TypeVar" but I'll check it. Thanks for the good idea.

Reply:
I found the solution but am not happy with defining is as a `T` variable. I tried to rename it to `WithAttribute` but it looks even worse :-/ Any idea for the TypeVar name and placement?

Reply:
You dislike it because then it doesn't show the union which actually describes the types?  I would agree that's a downside...  To make sure we're talking about the same thing.

https://play-mypy.ymyzk.com/?gist=7f1c24aa9bb97e724c1399ec61a79523
```python3
import typing

WithAttribute = typing.TypeVar('WithAttribute', int, str)

def f(x, y): # type: (WithAttribute, WithAttribute) -> None
    return

f(1, 1)
f('', '')
f(1, '')
```
```python
main.py:10: error: Value of type variable "WithAttribute" of "f" cannot be "object"
```
![image](https://user-images.githubusercontent.com/543719/54044821-e938de80-419d-11e9-8434-ec6b1754ef8b.png)

I'm guessing I'll go with whatever if we get through the rest and I haven't managed to reach my own opinion.

Reply:
My hesitation was only about the typevar name, if it is better "T" or "WithAttribute". The rest is clear.